client/src/components/RelatedAndComparison/RelatedAndComparison.jsx

```diff
@@ -1,16 +1,123 @@
-import React from 'react';
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-plusplus */
+import React, { useEffect, useState } from 'react';
 import axios from 'axios';
-import styled from 'styled-components';
-
-const Wrapper = styled.section`
-margin-top: 15em;
-padding: 20em;
-background: #DF9EF6;`;
-
-const RelatedAndComparison = () => (
-  <Wrapper>
-    <h2>Related Products</h2>
-  </Wrapper>
-);
+import RelatedProducts from './assets/RelatedProducts.jsx';
+import YourOutfit from './assets/YourOutfit.jsx';
+
+const RelatedAndComparison = () => {
+  const [relatedIds, setRelatedIds] = useState([]);
+  const [relatedStyles, setRelatedStyles] = useState([]);
+  const [relatedProducts, setRelatedProducts] = useState([]);
+  const [styles, setStyles] = useState([]);
+  const [yourOutfit, setYourOutfit] = useState([]);
+
+  useEffect(() => {
+    axios.get('/api/products/48432/related')
+      .then(({ data }) => {
+        setRelatedIds(data);
+      });
+  }, []);
+
+  useEffect(() => {
+    const relatedStylesData = relatedIds.map((item) => axios.get(`/api/products/${item}/styles`)
+      .then(({ data }) => {
+        const relatedStylesFormat = { id: data.product_id, results: [] };
+        const results = data.results;
+
+        for (let i = 0; i < results.length; i++) {
+          const resultsIdx = results[i];
+          const thisIn = {
+            salePrice: resultsIdx.sale_price,
+            originalPrice: resultsIdx.original_price,
+            images: resultsIdx.photos,
+          };
+          relatedStylesFormat.results.push(thisIn);
+        }
+        return relatedStylesFormat;
+      }));
+    Promise.all(relatedStylesData)
+      .then((values) => {
+        setRelatedStyles(values);
+      });
+  }, [relatedIds]);
```

Comment:
Good use of Promise.all here. During optimization, we may be able to raise these getRequests to the app level